File: spring-data-requery-java/src/main/java/com/coupang/springframework/data/requery/repository/config/RequeryRepositoryNameSpaceHandler.java
package com.coupang.springframework.data.requery.repository.config;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.xml.NamespaceHandlerSupport;
import org.springframework.data.repository.config.RepositoryBeanDefinitionParser;

/**
 * com.coupang.springframework.data.requery.repository.config.RequeryRepositoryNameSpaceHandler
 *
 * @author debop
 * @since 18. 6. 6
 */
@Slf4j
public class RequeryRepositoryNameSpaceHandler extends NamespaceHandlerSupport {

    @Override
    public void init() {
        log.debug("Init and regist BeanDefinitionParser for repositories.");

        RequeryRepositoryConfigurationExtension extension = new RequeryRepositoryConfigurationExtension();
        RepositoryBeanDefinitionParser definitionParser = new RepositoryBeanDefinitionParser(extension);

        registerBeanDefinitionParser("repositories", definitionParser);

        // auditing 은 필요없다.
        // registerBeanDefinitionParser("auditing", new AuditingBeanDefinitionParser());
    }
}

File: spring-data-requery-kotlin/src/main/kotlin/org/springframework/data/requery/kotlin/repository/config/RequeryRepositoryNameSpaceHandler.kt
package org.springframework.data.requery.repository.config

import mu.KotlinLogging.logger
import org.springframework.beans.factory.xml.NamespaceHandlerSupport
import org.springframework.data.repository.config.RepositoryBeanDefinitionParser

/**
 * org.springframework.data.requery.repository.config.RequeryRepositoryNameSpaceHandler
 *
 * @author debop
 */
class RequeryRepositoryNameSpaceHandler: NamespaceHandlerSupport() {

    private val log = logger { }

    override fun init() {
        log.debug { "Init and regist BeanDefinitionParser for repositories." }

        val extension = RequeryRepositoryConfigurationExtension()
        val definitionParser = RepositoryBeanDefinitionParser(extension)

        registerBeanDefinitionParser("repositories", definitionParser)

        // auditing 은 필요없다.
        // registerBeanDefinitionParser("auditing", new AuditingBeanDefinitionParser());
    }
}